### ISSUE TITLE
JSON file containing directions for cable_helper.py was not readable

### ISSUE THREAD

**Issue by rhliang** [cfe-lab/Kive#668]
The `cable_helper.py` (and `step_helper.py`) scripts depend on receiving a JSON file with instructions on what to do from the Manager.  We had a run fail in the last round of Octomore stress testing because the Slurm task that was to carry out the cable helper was unable to find the JSON file: 

```(vkive) [kivefleet@octomore logs]$ cat output1_stderr.txt 
Traceback (most recent call last):
  File "/var/spool/slurmd/job186404/slurm_script", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/home/kivefleet/vkive/lib/python2.7/site-packages/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/home/kivefleet/vkive/lib/python2.7/site-packages/django/core/management/__init__.py", line 345, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/kivefleet/vkive/lib/python2.7/site-packages/django/core/management/base.py", line 348, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/kivefleet/vkive/lib/python2.7/site-packages/django/core/management/base.py", line 399, in execute
    output = self.handle(*args, **options)
  File "/usr/local/share/Kive/kive/fleet/management/commands/cable_helper.py", line 16, in handle
    with open(options["cable_execution_info_json"], "rb") as f:
IOError: [Errno 2] No such file or directory: '/data/kive/Sandboxes/userkive_run5200_OPvXtl/step8/cable_infopc21SR.json'
```

and another one failed similarly.  To compensate, introduce some defensive code (i.e. `file_access_utils.confirm_file_created`) into the helper scripts so that it is more patient when waiting for the file with instructions.

**Comment by donkirkby** [comment]
Even with the defensive code, `cable_helper.py` failed when it couldn't read the JSON file.

    2017-04-09 11:21:36[WARNING]file_access_utils.confirm_file_created(): File /data/kive/Sandboxes/usermicall_run247674_RBnLNE/step5/cable_infoC53r76.json appears not to have been fully created yet; waiting 10.209834 seconds before retrying.
    2017-04-09 11:21:52[WARNING]file_access_utils.confirm_file_created(): File /data/kive/Sandboxes/usermicall_run247674_RBnLNE/step5/cable_infoC53r76.json appears not to have been fully created yet; waiting 11.490967 seconds before retrying.
    2017-04-09 11:22:07[WARNING]file_access_utils.confirm_file_created(): File /data/kive/Sandboxes/usermicall_run247674_RBnLNE/step5/cable_infoC53r76.json appears not to have been fully created yet; waiting 8.566976 seconds before retrying.
    2017-04-09 11:22:18[WARNING]file_access_utils.confirm_file_created(): File /data/kive/Sandboxes/usermicall_run247674_RBnLNE/step5/cable_infoC53r76.json appears not to have been fully created yet; waiting 11.657927 seconds before retrying.
    2017-04-09 11:22:32[WARNING]file_access_utils.confirm_file_created(): File /data/kive/Sandboxes/usermicall_run247674_RBnLNE/step5/cable_infoC53r76.json appears not to have been fully created yet; waiting 8.153565 seconds before retrying.
    Traceback (most recent call last):
      File "/var/spool/slurmd/job585707/slurm_script", line 10, in <module>
        execute_from_command_line(sys.argv)
      File "/home/kivefleet/vkive/lib/python2.7/site-packages/django/core/management/__init__.py", line 353, in execute_from_command_line
        utility.execute()
      File "/home/kivefleet/vkive/lib/python2.7/site-packages/django/core/management/__init__.py", line 345, in execute
        self.fetch_command(subcommand).run_from_argv(self.argv)
      File "/home/kivefleet/vkive/lib/python2.7/site-packages/django/core/management/base.py", line 348, in run_from_argv
        self.execute(*args, **cmd_options)
      File "/home/kivefleet/vkive/lib/python2.7/site-packages/django/core/management/base.py", line 399, in execute
        output = self.handle(*args, **options)
      File "/usr/local/share/Kive/kive/fleet/management/commands/cable_helper.py", line 27, in handle
        file_access_utils.confirm_file_created(options["cable_execution_info_json"])
      File "/usr/local/share/Kive/kive/file_access_utils.py", line 297, in confirm_file_created
        raise e
    file_access_utils.FileCreationError: File /data/kive/Sandboxes/usermicall_run247674_RBnLNE/step5/cable_infoC53r76.json was not created; checked 5 times over 67.528992 seconds.

**Comment by donkirkby** [comment]
It looks like the fix for #679 was released in May as part of v0.10.3, and I don't see this error in any of the log files since then. I'm going to assume that the changes to the file name somehow avoided some file conflicts, and close the issue for now.
